akka-actor-typed-tests/src/test/scala/akka/actor/typed/internal/ActorSystemSpec.scala

```diff
@@ -69,21 +71,35 @@ class ActorSystemSpec extends WordSpec with Matchers with BeforeAndAfterAll with
 
     "terminate the guardian actor" in {
       val inbox = TestInbox[String]("terminate")
-      val sys = system(
-        Behaviors
-          .receive[Probe] {
-            case (_, _) => Behaviors.unhandled
-          }
-          .receiveSignal {
-            case (_, PostStop) =>
-              inbox.ref ! "done"
-              Behaviors.same
-          },
-        "terminate")
+      val sys = system(Behaviors.setup[Any] { _ =>
+        inbox.ref ! "started"
+        Behaviors.receiveSignal {
+          case (_, PostStop) =>
+            inbox.ref ! "done"
+            Behaviors.same
+        }
+      }, "terminate")
+
+      eventually {
+        inbox.hasMessages should ===(true)
+      }
+      inbox.receiveAll() should ===("started" :: Nil)
+
+      // now we know that the guardian has started, and should receive PostStop
       sys.terminate().futureValue
       inbox.receiveAll() should ===("done" :: Nil)
     }
 
+    "be able to terminate immediately" in {
```

Comment:
added this test, which was similar as the failing test, but we can't verify much more than that the ActorSystem can shutdown